### ISSUE TITLE
Fuzz testing

### ISSUE THREAD

**Issue by mkg20001** [ipfs/js-ipfs#1295]
- **Version**: hopefully the next
- **Platform**: *
- **Subsystem**: *

#### Description: Fuzz Testing

I have discovered many critical application crashes by accident after my node was crashed by the invalid data it recived.

Such kinds of crashes are likely unacceptable in production.
Discovering them by accident isn't really a solution to that. https://github.com/ipfs/js-ipfs/issues/1292 https://github.com/libp2p/js-libp2p/issues/179

So I propose to add some tests that just bombard the node with random semi-valid or invalid data. Aka fuzz testing.

**Comment by diasdavid** [comment]
//cc @fsdiogo

**Comment by fsdiogo** [comment]
Where do you guys suggest I start? I'm a bit lost to be honest.

**Comment by diasdavid** [comment]
@fsdiogo are you familiar with the concept of fuzz testing -- https://en.wikipedia.org/wiki/Fuzzing --? It will be powerful for your goal of https://github.com/ipfs/js-ipfs/issues/1325.

@pgte and @satazor should/might have a lot of ideas to help you here :)

**Comment by fsdiogo** [comment]
Yes I'm familiar with the concept, although have never implemented it.

My question was a bit more specific, should I start in `js-ipfs`, `libp2p`?

What do you mean when a node receives invalid data?

**Comment by lidel** [comment]
I will go on a limb here and say that the low hanging fruit is CLI and HTTP API.  It may be easier to reason about creating fuzzing tests for them plus user/developer experience will directly benefit from that.

Fuzzing around messaging between nodes should happen in libp2p project, I think.

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).

Please feel to reopen with any comments before 2023-06-05. We will do a final pass on reopened issues afterward (see https://github.com/ipfs/js-ipfs/issues/4336).

There are no plans to implement this in Helia yet, please feel free to create an issue there.
